common/src/main/java/com/microsoft/identity/common/internal/telemetry/TelemetryContext.java

```diff
@@ -92,19 +95,7 @@ void addDeviceInfo(@NonNull final Context context) {
         put(Device.MANUFACTURER, Build.MANUFACTURER);
         put(Device.MODEL, Build.MODEL);
         put(Device.NAME, Build.DEVICE);
-        try {
-            put(
-                    Device.ID,
-                    StringExtensions.createHash(
-                            Settings.Secure.getString(
-                                    context.getContentResolver(),
-                                    Settings.Secure.ANDROID_ID
-                            )
-                    )
-            );
-        } catch (final NoSuchAlgorithmException | UnsupportedEncodingException exception) {
-            Logger.warn(TAG, "Unable to get the device id.");
-        }
+        put(Device.ID, mTelemetryPropsCache.getOrCreateRandomStableDeviceId());
```

Comment:
Why not the ad-id?

Reply:
Ad-ID is just a GUID, not sure why we would add a library dependency on Play Services for this...
https://developers.google.com/android/reference/com/google/android/gms/ads/identifier/AdvertisingIdClient

Reply:
You get two things out of it.  The ad-id is actually per-device, and it is resettable by the user.  If the concern is privacy, then that's better than the per-install flag.

Reply:
This is in support of a public telemetry API.  I think that an application bound GUID is the right approach, but we should be VERY EXPLICIT about it in the documentation so that people aren't confused.

Reply:
Synced offline, will use a generated GUID